crates/ruff_linter/src/source_kind.rs

```diff
@@ -154,7 +159,7 @@ impl SourceKind {
 
                 Ok(())
             }
-            _ => bail!("cannot diff Python source code with Jupyter notebook source code"),
+            _ => panic!("cannot diff Python source code with Jupyter notebook source code"),
```

Comment:
We could potentially create a new `SourceError::IncompatibleDiffSource` variant but then the `Diagnostics:from_source_error` needs to be updated which I don't think is correct because the diff error shouldn't really be a diagnostic.

Reply:
I really wish we had a way to enforce this statically through types (i.e., that this path should never be reached).